core/tests/test_runner/test_file.rs

```diff
@@ -0,0 +1,71 @@
+use std::path::{Path, PathBuf};
+
+use serde::{Deserialize, Serialize};
+
+// TODO: replace with Config?
+#[derive(Deserialize, Clone, Serialize)]
+pub struct Preamble {
+    pub title: Option<String>,
+    pub author: Option<String>,
+    pub lang: Option<String>,
+}
+
+#[derive(Deserialize)]
+pub struct TestOutputs {
+    pub html: Option<String>,
+    pub um: Option<String>,
+}
+
+#[derive(Deserialize)]
+pub struct Test {
+    pub name: String,
+    pub description: Option<String>,
+    pub preamble: Option<Preamble>,
+    pub input: String,
+
+    #[serde(flatten)]
+    pub outputs: TestOutputs,
+}
+
+#[derive(Deserialize)]
+pub struct TestFile {
+    pub spec: String,
+    pub name: String,
+    pub description: Option<String>,
+    pub preamble: Option<Preamble>,
+    pub tests: Vec<Test>,
+}
+
+pub struct TestContent {
+    pub test_file: TestFile,
+    pub snap_path: PathBuf,
+}
+
+pub fn get_test_content(test_filepath: &str) -> TestContent {
+    let mut path = PathBuf::from(env!("CARGO_MANIFEST_DIR"))
+        .canonicalize()
+        .unwrap();
+    path.push("tests/");
+    let sanitized_filepath = if test_filepath.starts_with(|c| c == '/' || c == '\\') {
+        &test_filepath[1..]
+    } else {
+        test_filepath
+    };
```

Comment:
Just a thought, it might happen that we provide path like `////some/path`. In that case skipping only one `/` is not doing much, but we would have to be crazy to accept PR containing test with such path 🤣

Reply:
Skipping one `/` helps to accept "absolut" paths set by contributors not carefully reading doc-comments.
I myself often wonder if a path must be given with leading `/` or not. So I made the macro accept both 😄

Reply:
That's fine, I just meant that `///some/path` will be changed to `//some/path` by this code. But like already said, that would be really weird thing to do anyway and should be caught in PR review.